TSC/2021/04-30.md

```diff
@@ -0,0 +1,81 @@
+
+ACTION: AW to add relevant issues on GitHub.
+
+- AM Asked for clarification on the difference between Docker and Snap, to which AW explained that Snap doesn't use a VM, whereas Docker does, and that from the user's perspective a Snap acts like a single binary container, and which may be potentially stored on a central repository allowing automatic package updates on user's machines.
+
+## 2. Parallelism
+
+- SS stated that he was currently working on multithreading force calculation having completed related work on energy calculation.
+- SS questions some specifics behind potentially shared class variables when calculating torsions and impropers.
+- TY stated that these calculations were amongst the worst (in terms of equations and variable storage requirements) in Dissolve, but that the variables could be process independent provided one torsion/improper is calculated by one process.
+- SS suggested further ways of improving the use of `ForceKernel` in multithreaded applications
+- TY asked what the plan going forward into the next sprint would be, suggesting that once forces were complete that it might be nice to implement baseline MPI/threads algorithms for exemplar functions such as RDF and energy calculation, leaving the more difficult problems (e.g. Monte Carlo) until the approach is established.
+- SS and AW agreed.
+
+## 3. Benchmarking
+
+- TY noted that, aside from the GitHub pages not directly displaying the benchmarks because they are not linked to, there was a performance regression flagged by the system from a commit which did not in itself change any code.
+- SS suggested that this was most likely just a "blip" on the instance running the benchmarks, to which TY agreed.
+
+## 4. GUI - Charts
+
+- TY raised a question over the viability of the current custom visualisation (viewer) systems present within Dissolve, citing that they are "a bit ugly", not as performant as desired, are responsible for several tracked bugs, and require a fragile dependency in FTGL.
+- TY considered the possibility of using QtChart as a replacement, and asked for the opinions of the team.
+- None of the team raised and specific concerns about using QtChart, but wondered about any alternatives.
+- It was mentioned that DN had previously search for device- / platform-agnostic rendering solutions for projects he was involved in, but TY recalled that he had eventually gone back to VTK, which is a significant dependency to add in to Dissolve.
+- AW further suggested that the OpenGL-based species / configuration rendering was simply balls and cylinders, and could probably be easily realised using Qt3D.
+- TY agreed that this was certainly worth doing as part of the transition to Qt-based tech.
+- SS raised the issue that the Qt company seem to be on a drive towards commercial licenses only.
```

Comment:
One thing to look at reasonably soon would be moving to Qt6 given that Qt5 is now commercial only support.